src/result_processor.c

```diff
@@ -786,8 +790,12 @@ static SearchResult *GetNextResult(RPSafeLoader *self) {
 static int rpSafeLoaderNext_Accumulate(ResultProcessor *rp, SearchResult *res);  // Forward declaration
 
 static int rpSafeLoader_ResetAndReturnLastCode(RPSafeLoader *self) {
-
-  self->base_loader.base.Next = rpSafeLoaderNext_Accumulate; // Reset the next function, in case we are in cursor mode
+  // Reset the next function, in case we are in cursor mode
+  if (self->becomePlainLoader) {
+    self->base_loader.base.Next = rploaderNext;
```

Comment:
We can, but it requires too much effort to improve the mock and there is no way of choosing the first option today with any query. It would become relevant only if we want to change the SL accumulation logic, and then we should also cover this case